static/app/components/superuserAccessForm.tsx

```diff
@@ -21,31 +22,53 @@ type OnTapProps = NonNullable<React.ComponentProps<typeof U2fContainer>['onTap']
 
 type Props = {
   api: Client;
+  hasStaff: boolean;
 };
 
 type State = {
   authenticators: Array<Authenticator>;
   error: boolean;
   errorType: string;
+  isLoading: boolean;
   showAccessForms: boolean;
   superuserAccessCategory: string;
   superuserReason: string;
 };
 
 class SuperuserAccessForm extends Component<Props, State> {
-  state: State = {
-    authenticators: [],
-    error: false,
-    errorType: '',
-    showAccessForms: true,
-    superuserAccessCategory: '',
-    superuserReason: '',
-  };
+  constructor(props) {
+    super(props);
+    this.authUrl = this.props.hasStaff ? '/staff-auth/' : '/auth/';
+    this.state = {
+      authenticators: [],
+      error: false,
+      errorType: '',
+      showAccessForms: true,
+      superuserAccessCategory: '',
+      superuserReason: '',
+      isLoading: true,
+    };
+  }
+
+  async componentDidMount() {
+    const disableU2FForSUForm = ConfigStore.get('disableU2FForSUForm');
 
-  componentDidMount() {
-    this.getAuthenticators();
+    // If using staff and on local dev, skip U2F and immediately submit
+    if (this.props.hasStaff && disableU2FForSUForm) {
+      await this.handleSubmit(this.state);
+      return;
+    }
+
+    await this.getAuthenticators();
+    // Set the error state if there are no authenticators and U2F is on
+    if (!this.state.authenticators.length && !disableU2FForSUForm) {
+      this.handleError(ErrorCodes.NO_AUTHENTICATOR);
+    }
```

Comment:
should this go inside `this.getAuthenticators`? we already set `this.state.authenticators.length` to `[]` in there